exporter/dynatraceexporter/internal/serialization/sum_test.go

```diff
@@ -197,163 +198,201 @@ func Test_serializeSum(t *testing.T) {
 		assert.ElementsMatch(t, makeSimplifiedLogRecordsFromObservedLogs(observedLogs), expectedLogs)
 	})
 
-	t.Run("monotonic delta is exported as delta", func(t *testing.T) {
+	t.Run("monotonic delta", func(t *testing.T) {
 		metric := pmetric.NewMetric()
 		metric.SetDataType(pmetric.MetricDataTypeSum)
 		metric.SetName("metric_name")
 		sum := metric.Sum()
 		sum.SetAggregationTemporality(pmetric.MetricAggregationTemporalityDelta)
 		sum.SetIsMonotonic(true)
 
-		// not checking Double, this is done in Test_serializeSumPoint
-		validDp := sum.DataPoints().AppendEmpty()
-		validDp.SetIntVal(12)
+		dp := sum.DataPoints().AppendEmpty()
+		t.Run("with valid value is exported as delta", func(t *testing.T) {
+			// not checking Double, this is done in Test_serializeSumPoint
+			dp.SetIntVal(12)
 
-		invalidDp := sum.DataPoints().AppendEmpty()
-		invalidDp.SetDoubleVal(math.NaN())
+			prev := ttlmap.New(10, 10)
 
-		prev := ttlmap.New(10, 10)
+			zapCore, observedLogs := observer.New(zap.WarnLevel)
+			logger := zap.New(zapCore)
 
-		zapCore, observedLogs := observer.New(zap.WarnLevel)
-		logger := zap.New(zapCore)
+			actualLines := serializeSum(logger, "", metric, empty, empty, prev, []string{})
 
-		actualLines := serializeSum(logger, "", metric, empty, empty, prev, []string{})
+			expectedLines := []string{
+				"metric_name count,delta=12",
+			}
+			assert.ElementsMatch(t, actualLines, expectedLines)
+			assert.Empty(t, observedLogs.All())
+		})
 
-		expectedLines := []string{
-			"metric_name count,delta=12",
-		}
-		expectedLogRecords := []simplifiedLogRecord{
-			{
-				message: "Error serializing sum data point",
-				attributes: map[string]string{
-					"name":       "metric_name",
-					"value-type": "Double",
-					"error":      "value is NaN.",
+		t.Run("with invalid value logs warning and returns no line", func(t *testing.T) {
```

Comment:
Ah this is a cool approach with the nested tests. Nice!

Reply:
Yeah, it's pretty cool actually!